virttest/ceph.py

```diff
@@ -220,3 +221,74 @@ def create_config_file(ceph_monitor):
         with open(ceph_cfg, 'w+') as f:
             f.write('mon_host = %s' % ceph_monitor)
     return ceph_cfg
+
+
+@error_context.context_aware
+def cephfs_mount(ceph_uri, mount_point, options=None, verbose=False, session=None):
+    """
+    Mount ceph volume to mountpoint.
+
+    :param ceph_uri: str to express ceph uri, e.g 10.73.xx.xx:6789:/
+    :param mount_point: mount point, str
+    :param options: mount options, e.g -o name=admin,secret=AQCAgd5cFyMQLhAAHaz6w+WKy5LvmKjmxx
+    :param verbose: enable verbose or not
+    :param session: mount within the session if given
+    """
+    cephfs_umount(ceph_uri, mount_point, verbose, session)
+    mount_cmd = ['mount -t ceph']
+    if options:
+        mount_cmd.extend(['-o', options])
+    mount_cmd.extend([ceph_uri, mount_point])
+    mount_cmd = ' '.join(mount_cmd)
+    if session:
+        create_mount_point = "mkdir -p  %s" % mount_point
+        session.cmd(create_mount_point, ok_status=[0, 1], ignore_all_errors=True)
+        session.cmd(mount_cmd, ok_status=[0, 1], ignore_all_errors=True)
+    else:
+        if not os.path.exists(mount_point):
+            try:
+                os.mkdir(mount_point)
+            except OSError as dirError:
+                logging.debug("Creation of the directory:%s failed:%s", mount_point, str(dirError))
+            else:
+                logging.debug("Successfully created the directory %s", mount_point)
+        process.system(mount_cmd, verbose=verbose)
+
+
+@error_context.context_aware
+def cephfs_umount(ceph_uri, mount_point, verbose=False, session=None):
+    """
+    Umount ceph volume from mountpoint.
+
+    :param ceph_uri: str to express ceph uri, e.g 10.73.xx.xx:6789:/
+    :param mount_point: mount point, str
+    :param verbose: enable verbose or not
+    :param session: mount within the session if given
+    """
+    umount_cmd = "umount %s" % mount_point
+    if session:
+        session.cmd(umount_cmd, ok_status=[0, 1], ignore_all_errors=True)
+        rm_mount_point = "rm -rf %s" % mount_point
+        session.cmd(rm_mount_point, ok_status=[0, 1], ignore_all_errors=True)
+    else:
+        process.system(umount_cmd, ignore_status=True, verbose=verbose)
+        if os.path.exists(mount_point):
+            try:
+                os.rmdir(mount_point)
+            except OSError as dirError:
+                logging.debug("Delete of the directory:%s failed:%s", mount_point, str(dirError))
+            else:
+                logging.debug("Successfully deleted the directory %s", mount_point)
```

Comment:
Can you check utils_disk.safe_rmdir and utils_misc.umount. you can reuse them I think.

Reply:
I assume there is no safe_rmdir in utils_disk.py, but in utils_mic.py.  Actually, The method has quite limited usability, it can not handle the scenario gracefully where directory not existed.
But it can have a try.
*******************************************
>>> utils_misc.safe_rmdir('/tmp/abc')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/chunfu/workdir/p3-v3/code/avocado-vt/virttest/utils_misc.py", line 993, in safe_rmdir
    assert check_isdir(path, session=session), "Invalid directory to remove %s" % path
AssertionError: Invalid directory to remove /tmp/ab
*******************************************************
utils_misc.umount call utils_disk.umount() underlying, which has issue as mentioned previously